internal/setup/yaml.go

```diff
@@ -373,35 +404,74 @@ func parseRelease(baseDir, filePath string, data []byte) (*Release, error) {
 	return release, err
 }
 
-func parsePackage(baseDir, pkgName, pkgPath string, data []byte) (*Package, error) {
+func parsePackage(format, baseDir, pkgName, pkgPath string, data []byte) (*Package, error) {
```

Comment:
side note: This function is very big, and got even bigger in this PR. A refactoring PR after this one to split this function might help make it easier to manage.